include/triton/common/logging.h

```diff
@@ -33,46 +33,54 @@
 #include <string>
 #include <vector>
 
-namespace triton { namespace common {
-
-// A log message.
-class LogMessage {
- public:
-  // Log levels.
-  enum Level { kERROR = 0, kWARNING = 1, kINFO = 2 };
+#include "table_printer.h"
+#ifdef _WIN32
+// suppress the min and max definitions in Windef.h.
+#define NOMINMAX
+#include <Windows.h>
+#else
+#include <sys/time.h>
+#include <sys/types.h>
+#include <time.h>
+#include <unistd.h>
+#endif
 
-  LogMessage(const char* file, int line, uint32_t level);
-  ~LogMessage();
 
-  std::stringstream& stream() { return stream_; }
+namespace triton { namespace common {
 
- private:
-  static const std::vector<char> level_name_;
-  std::stringstream stream_;
-};
 
 // Global logger for messages. Controls how log messages are reported.
 class Logger {
  public:
+  // Log Formats.
   enum class Format { kDEFAULT, kISO8601 };
 
+  // Log levels.
+  enum Level { kERROR = 0, kWARNING = 1, kINFO = 2, kEND };
+
+  inline static const std::array<const char*, Level::kEND> LEVEL_NAMES{
+      "E", "W", "I"};
+
   Logger();
 
   // Is a log level enabled.
-  bool IsEnabled(LogMessage::Level level) const { return enables_[level]; }
+  bool IsEnabled(Level level) const { return enables_[level]; }
 
   // Set enable for a log Level.
-  void SetEnabled(LogMessage::Level level, bool enable)
-  {
-    enables_[level] = enable;
-  }
+  void SetEnabled(Level level, bool enable) { enables_[level] = enable; }
 
   // Get the current verbose logging level.
   uint32_t VerboseLevel() const { return vlevel_; }
 
   // Set the current verbose logging level.
   void SetVerboseLevel(uint32_t vlevel) { vlevel_ = vlevel; }
 
+  // Whether to escape log messages
+  // using JSON string escaping rules.
+  // Default is true but can be disabled via an environment variable
+  // TRITON_SERVER_ESCAPE_LOG_MESSAGES
```

Comment:
or "=FALSE" pending other comment, but nice to show example usage somewhere.
```suggestion
  // Default is true but can be disabled via environment variable:
  // TRITON_SERVER_ESCAPE_LOG_MESSAGES=0
```